src/cuda/api/device.hpp

```diff
@@ -440,9 +440,9 @@ class device_t {
 	 */
 	device::pci_location_t pci_id() const
 	{
-		auto pci_domain_id = get_attribute(cudaDevAttrPciDomainId);
-		auto pci_bus_id = get_attribute(cudaDevAttrPciBusId);
-		auto pci_device_id = get_attribute(cudaDevAttrPciDeviceId);
+		auto pci_domain_id = get_attribute(static_cast<attribute_t>(cudaDevAttrPciDomainId));
```

Comment:
Again, need to look at the `get_attribute()` function itself rather than cast its results all the time.